### PR TITLE
chore: upgrade faker-js latest

### PR DESCRIPTION
**Pull request by avallete**
- Upgrade faker to latest version
- Allow `generateHashKey` to use arbirary long seed string

### CODE REVIEW

File: src/hash.ts [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+  return output.padEnd(16, '0')
+}
+
 export const setHashKey = hash.setKey
 
-export const generateHashKey = hash.generateKey
+export const generateHashKey = (input: string) => {
+  return input.length === 16
+    ? hash.generateKey(input)
+    : hash.generateKey(derive16CharacterString(input))
```
Comment (justinvdm) [inline src/hash.ts:27]: Nice!

**Review by justinvdm — APPROVED** [review]
🍰! Thank you @avallete, this looks awesome. Just want to double check we're ok with the breaking change, I don't think there's a way to avoid it, but I know it's been something we've tried to avoid before. I'll ping you on discord though and we can talk there.

**Comment by avallete** [comment]
Alright after some discussion we decided to:

1. Import directly the definitions from faker to not break determinism with older versions
2. Maybe in the future completely delete faker as we only use it to re-export now

@justinvdm I also had to write a custom script for esbuild to build recursively the sources inside `src/locales/...`

If you're happy with it we can merge and release the new version on npm, we should not have any breaking changes.

File: scripts/buildjs.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const inputDir = './src';
+const outputDir = './dist';
+
+// Find all '.ts' files in inputDir (recursively), excluding '.test.ts' files
+const files = glob.sync(`${inputDir}/**/*.ts`, { ignore: `${inputDir}/**/*.test.ts` });
+
+const options = {
+  format: 'cjs',
+  // Add any other options you need
```
Comment (justinvdm) [inline scripts/buildjs.js:13]: Do we also need to copy over `--platform=node --target=node14` to here?

Comment (avallete) [reply scripts/buildjs.js:13]: Ouhh yeah probably nice spot

**Comment by justinvdm** [comment]
Thank you for doing this @avallete! Looks great, happy for it to land 👍

**Pull request merged**